lib/producer/higher-level-producer.js

```diff
@@ -0,0 +1,201 @@
+/*
+ * node-rdkafka - Node.js wrapper for RdKafka C/C++ library
+ *
+ * Copyright (c) 2016 Blizzard Entertainment
+ *
+ * This software may be modified and distributed under the terms
+ * of the MIT license.  See the LICENSE.txt file for details.
+ */
+
+module.exports = HighLevelProducer;
+
+var util = require('util');
+var Producer = require('../producer');
+var LibrdKafkaError = require('../error');
+var EventEmitter = require('events').EventEmitter;
+var RefCounter = require('../tools/ref-counter');
+var shallowCopy = require('../util').shallowCopy;
+var isObject = require('../util').isObject;
+
+util.inherits(HighLevelProducer, Producer);
+
+function noopSerializer(v) {
+  return v;
+}
+
+/**
+ * Create a serializer
+ *
+ * Method simply wraps a serializer provided by a user
+ * so it adds context to the error
+ *
+ * @returns {function} Serialization function
+ */
+function createSerializer(serializer) {
```

Comment:
Should this support async serializers? When initializing a `Producer` or `HighLevelProducer` you aren't required to pre-define your topics, which means that any serializer that requires some kind of async action would have to happen in the flow of the `produce` call. An example of this would be Confluent's Schema Registry service.

Reply:
Sure. I think that makes sense. We can make them support a callback, or potentially do backwards compatible checking for promise methods and utilize those.